Refactor part of padding logic

AppearamidGuy opened the pull request:
This pr refactors part of padding selection logic to improve consistency with the original design of the app, fix bugs, and remove hacky parts. I've added short explanations of the changes to the commits but all you really need to know is in this link: [Why would I want to fitsSystemWindows](https://medium.com/androiddevelopers/why-would-i-want-to-fitssystemwindows-4e26d9ce1eec)

This fixes #34 and a couple other visual bugs. However, I'm publishing it as a draft because I have some questions and I tested only some combinations of settings. I am considering adding automated screenshots (e.g. using fastlane) to simplify testing all the combinations. Also I've mostly skipped comments screen because it is the most complicated one and it's not obvious if it can be simplified.

AppearamidGuy commented on app/src/main/res/layout/activity_about.xml:
```diff
@@ -13,7 +13,8 @@
         android:layout_width="match_parent"
-        android:layout_height="match_parent">
+        android:layout_height="match_parent"
+        android:fitsSystemWindows="true">
```
Here and in other places that don't set `clipToPadding="false"`, content won't scroll under navigation bar. Keep it as is or change it to make consistent with main screen?

SimonHalvdansson replied:
This should be clip ToPadding="false" I believe - right now the last two buttons at the very end of the submissions list can be hard to press.

AppearamidGuy replied:
You're thinking about padding itself, `fitsSystemWindows` will take care of that already.
`clipToPadding` controls how recycler displays its content when it is not completely scrolled to one end. By default (`clipToPadding=true`) it will not display anything in padding area, i.e. clip its content. With `clipToPadding=false`, because padding is equal to system bars in this case, recycler will display content under system bars when not completely scrolled and empty space when completely scrolled

AppearamidGuy commented on app/src/main/java/com/simon/harmonichackernews/MainActivity.java:
```diff
@@ -109,11 +109,7 @@ private void updateFragmentLayout() {
-            }
+            findViewById(R.id.main_fragments_container).setPadding(extraPadding, 0, 0, 0);
         }
     }
```
Is there a reason you're trying to handle configuration changes yourself?

AppearamidGuy commented on app/src/main/java/com/simon/harmonichackernews/StoryRecyclerViewAdapter.java:
```diff
@@ -508,11 +504,6 @@ public class SubmissionsHeaderViewHolder extends RecyclerView.ViewHolder {
         public SubmissionsHeaderViewHolder(View view) {
```
Usually headers are placed in Toolbars that [collapse on scroll](https://github.com/material-components/material-components-android/blob/master/docs/components/TopAppBar.md#applying-scrolling-behavior-to-the-top-app-bar). Might be worth refactoring later

AppearamidGuy commented on app/src/main/java/com/simon/harmonichackernews/SubmissionsActivity.java:
Most of remaining usages of `Utils.getStatusBarHeight` are for setting position of progress indicator on `swipeRefreshLayout` like here. I think that's the only usecase when behavior should depend on transparency of the status bar. 
Right now the behavior is definitely wrong because indicator doesn't go all the way off screen with transparent status bar. A fix that makes the most sense to me is this, so indicator always starts just outside the visible screen and moves the same distance:
```
int offset = Utils.shouldUseTransparentStatusBar(this) ? 0 : Utils.getStatusBarHeight(getResources());
swipeRefreshLayout.setProgressViewOffset(
                false,
                swipeRefreshLayout.getProgressViewStartOffset() + offset,
                swipeRefreshLayout.getProgressViewEndOffset() + offset);
```
But it can be almost anything you want as long as `ProgressViewStartOffset` is not changed when status bar is transparent.

cc @fireph because you didn't like original behavior

AppearamidGuy commented:
Added fix for #42

SimonHalvdansson commented:
This looks very nice and comprehensive and publishing as a draft first is probably best as you say because these things are very phone to introduce bugs.

It's very nice of you to go through all this work, this really sets the passing logic on the right path again and I fully agree with the idea of the commit.

I am on another short vacation and the first week of August I'll be at a conference so I won't be able to take a look at this (and compile it to try out) until then so I will need to come back later with some in depth comments.

Again thanks for your hard work and I look forward to merging this so I can feel better about the padding handling throughout the app 😀

AppearamidGuy commented:
Rebased to the latest main and changed position of SwipeRefreshLayout's progress view to what I think is the correct one. There are still some places that don't use insets/fitsSystemWindows, but the overall state of the app is defenitely better in terms of both visuals and code quality.

@SimonHalvdansson have you had a chance to look at this pr?
